src/Build/Evaluation/Conditionals/MultipleComparisonExpressionNode.cs

```diff
@@ -48,36 +48,49 @@ internal override bool BoolEvaluate(ConditionEvaluator.IConditionEvaluationState
             // and we know which do, then we already have enough information to evaluate this expression.
             // That means we don't have to fully expand a condition like " '@(X)' == '' " 
             // which is a performance advantage if @(X) is a huge item list.
-            if (LeftChild.EvaluatesToEmpty(state) || RightChild.EvaluatesToEmpty(state))
+            bool leftEmpty = LeftChild.EvaluatesToEmpty(state);
+            bool rightEmpty = RightChild.EvaluatesToEmpty(state);
+            if (leftEmpty || rightEmpty)
             {
                 UpdateConditionedProperties(state);
 
-                return Compare(LeftChild.EvaluatesToEmpty(state), RightChild.EvaluatesToEmpty(state));
+                return Compare(leftEmpty, rightEmpty);
             }
 
-            if (LeftChild.CanNumericEvaluate(state) && RightChild.CanNumericEvaluate(state))
+            double? leftNumericValue = LeftChild.TryNumericEvaluate(state);
+            if (leftNumericValue is not null)
             {
-                return Compare(LeftChild.NumericEvaluate(state), RightChild.NumericEvaluate(state));
+                // The left child evaluating to a number and the right child not evaluating to a number
+                // is insufficient to say they are not equal because $(MSBuildToolsVersion) evaluates to
+                // the string "Current" most of the time but when doing numeric comparisons, is treated
+                // as a version and returns "17.0" (or whatever the current tools version is). This means
+                // that if '$(MSBuildToolsVersion)' is "equal" to BOTH '17.0' and 'Current' (if 'Current'
+                // is 17.0).
+                double? rightNumericValue = RightChild.TryNumericEvaluate(state);
+                if (rightNumericValue is not null)
+                {
+                    return Compare(leftNumericValue.Value, rightNumericValue);
+                }
```

Comment:
I don't know if there are other cases similar to MSBuildToolsVersion. I would guess not, at a glance. Maybe make this:
```suggestion
                double? rightNumericValue = RightChild.TryNumericEvaluate(state);
                if (rightNumericValue is not null)
                {
                    return Compare(leftNumericValue.Value, rightNumericValue);
                }
                else if (!LeftChild.DebuggerDisplay.Equals("$(MSBuildToolsVersion)")
                {
                        return false;
                }
```